File: src/C3.Catalogue/Decks/IDeckRepository.vb
Namespace Decks

    Public Interface IDeckRepository

        Function GetAll() As IList(Of Deck)
        Function FindByName(name As String) As Deck
        Function NameExists(name As String) As Boolean
        Function IsReferencedByTape(name As String) As Boolean
        Sub Add(value As Deck)
        Sub Update(value As Deck)
        Sub Delete(name As String)

    End Interface

End Namespace

File: src/C3.Catalogue/Decks/IDeckRepository.cs
using System.Collections.Generic;

namespace C3.Catalogue.Decks
{
    public interface IDeckRepository
    {
        IList<Deck> GetAll();

        Deck FindByName(string name);

        bool NameExists(string name);

        bool IsReferencedByTape(string name);

        void Add(Deck value);

        void Update(Deck value);

        void Delete(string name);
    }
}
